experimental/ActionPolyRing/src/Content.jl

```diff
@@ -0,0 +1,1258 @@
+###############################################################################
+#
+#  Construction 
+#
+###############################################################################
+
+##### Algebras #####
+
+### Difference ###
+@doc raw"""
+    difference_polynomial_ring(R::Ring, n_elementary_symbols::Int, ndiffs::Int) -> Tuple{DifferencePolyRing, Vector{DifferencePolyRingElem}}
+
+Return a tuple consisting of the difference polynomial ring over the ring `R` with the specified number of elementary variables and commuting endomorphisms, and the vector of
+these elementary variables. This methods allows for all the keywords of the `set_ranking!` method. See there for further information.
+
+# Examples
+
+```jldoctests
+julia> S, variables = difference_polynomial_ring(QQ, 3, 4)
+(Difference polynomial ring in 3 elementary symbols over QQ, DifferencePolyRingElem{QQFieldElem}[u1[0,0,0,0], u2[0,0,0,0], u3[0,0,0,0]])
+
+julia> S
+Difference polynomial ring in 3 elementary symbols u1, u2, u3
+with 4 commuting endomorphisms
+  over rational field
+
+julia> variables
+3-element Vector{DifferencePolyRingElem{QQFieldElem}}:
+ u1[0,0,0,0]
+ u2[0,0,0,0]
+ u3[0,0,0,0]
+```
+"""
+function difference_polynomial_ring(R::Ring, n_elementary_symbols::Int, ndiffs::Int; kwargs...)
+  dpr = DifferencePolyRing{elem_type(typeof(R))}(R, n_elementary_symbols, ndiffs)
+  set_ranking!(dpr; kwargs...)
+  return (dpr, deepcopy.(__add_new_jetvar!(dpr, [(i, zeros(Int, ndiffs)) for i in 1:n_elementary_symbols])))
+end
+
+@doc raw"""
+    difference_polynomial_ring(R::Ring, elementary_symbols::Vector{Symbol}, ndiffs::Int) -> Tuple{DifferencePolyRing, Vector{DifferencePolyRingElem}}
+
+Return a tuple consisting of the difference polynomial ring over the ring `R` with the specified elementary variables and number of commuting endomorphisms, and the vector of
+these elementary variables. Note that the multiindex [0..0] of length 'ndiffs' is appended to the variable names provided. This methods allows for all the keywords of the `set_ranking!`
```

Comment:
```suggestion
these elementary variables. Note that the multiindex `[0..0]` of length `ndiffs` is appended to the variable names provided. This methods allows for all the keywords of the `set_ranking!`
```
or something like this. otherwise this is interpreted as a markdown hyperlink